GNUmakefile

```diff
@@ -203,6 +203,9 @@ check: ## Lint the source code
 	@echo "==> Spell checking website..."
 	@misspell -error -source=text website/pages/
 
+	@echo "==> Checking for breaking changes in protos..."
```

Comment:
We should add the installation of `buf` to the `deps` target and the go.mod file in the `tools/` directory as well, so that we're pinning the version for folks who aren't using Vagrant.

Reply:
I've added buf to the tools, but to avoid a merge conflict I've skipped adding it to the current `deps` make target. This should be pulled in after https://github.com/hashicorp/nomad/pull/9357.